From 1c8ab0fce3cda29834cc5cd2c7dc99ad5bcc0a6b Mon Sep 17 00:00:00 2001
From: Jia Zhang <zhang.jia@linux.alibaba.com>
Date: Fri, 12 Oct 2018 14:11:04 +0800
Subject: [PATCH 3/3] Fix memory leak in grub_tpm1/2_log_event()

Signed-off-by: Jia Zhang <zhang.jia@linux.alibaba.com>
---
 grub-core/kern/efi/tpm.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/grub-core/kern/efi/tpm.c b/grub-core/kern/efi/tpm.c
index 9d8166a80..a2ded12a9 100644
--- a/grub-core/kern/efi/tpm.c
+++ b/grub-core/kern/efi/tpm.c
@@ -192,6 +192,7 @@ grub_tpm1_log_event(grub_efi_handle_t tpm_handle, unsigned char *buf,
   algorithm = TCG_ALG_SHA;
   status = efi_call_7 (tpm->log_extend_event, tpm, (grub_efi_physical_address_t)buf, (grub_uint64_t) size,
 		       algorithm, event, &eventnum, &lastevent);
+  grub_free (event);
 
   switch (status) {
   case GRUB_EFI_SUCCESS:
@@ -238,6 +239,7 @@ grub_tpm2_log_event(grub_efi_handle_t tpm_handle, unsigned char *buf,
 
   status = efi_call_5 (tpm->hash_log_extend_event, tpm, 0, (grub_efi_physical_address_t)buf,
 		       (grub_uint64_t) size, event);
+  grub_free(event);
 
   switch (status) {
   case GRUB_EFI_SUCCESS:
-- 
2.14.4.44.g2045bb6

